boskos/ranch/priority.go

```diff
@@ -0,0 +1,219 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ranch
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+// request stores request information with expiration
+type request struct {
+	id         string
+	expiration time.Time
+}
+
+func newRequestQueue() *requestQueue {
+	return &requestQueue{
+		requestMap: map[string]request{},
+	}
+}
+
+// requestQueue is a simple FIFO queue for requests.
+type requestQueue struct {
+	lock        sync.RWMutex
+	requestList []string
+	requestMap  map[string]request
+}
+
+// update updates expiration time is updated if already present,
+// add a new requestID at the end otherwise (FIFO)
+func (rq *requestQueue) update(requestID string, newExpiration time.Time) {
+	if requestID == "" {
+		// Nothing to do
+		return
+	}
+	rq.lock.Lock()
+	defer rq.lock.Unlock()
+	if rq.requestMap == nil {
```

Comment:
nit: do this in initialization/construction

Reply:
nice catch this was a leftover from some previous code. It is not needed anymore.

Reply:
done